plugins/collections/get.js

```diff
@@ -6,6 +6,42 @@ const schema = require('screwdriver-data-schema');
 const getSchema = schema.models.collection.get;
 const idSchema = joi.reach(schema.models.collection.base, 'id');
 
+/**
+ * Helper function to get last builds of pipelines
+ *
+ * @param {Pipeline} pipelines     - pipeline models to get last builds for
+ * @param {Factory}  eventFactory - factory for getting last event model
+ * @returns {Array}
+ */
+function getPipelinesHealth(pipelines, eventFactory) {
+    return pipelines
+        .filter(pipeline => !!pipeline)
+        .map((pipeline) => {
+            const result = Object.assign({}, pipeline.toJson());
+
+            if (!pipeline.lastEventId) {
+                return result;
+            }
+
+            return eventFactory.get(pipeline.lastEventId)
+                .then((event) => {
+                    if (!event) {
+                        return result;
+                    }
+
+                    return event.getBuilds()
+                        .then((builds) => {
+                            if (builds.length) {
+                                result.lastBuilds = builds.map(b => b.toJson());
```

Comment:
I was discussing it with @d2lam yesterday and we decided on showing all the builds for the last event rather than just a single build status to give more context. The reason I'm passing in the entire build objects here is to allow for sorting on most recent builds and for other stuff that might be needed in the future on the UI.